android/src/main/java/com/amplitude/android/Amplitude.kt

```diff
@@ -16,11 +18,21 @@ import com.amplitude.id.IdentityConfiguration
 import kotlinx.coroutines.launch
 
 open class Amplitude(
-    configuration: Configuration
-) : Amplitude(configuration) {
-
+    configuration: Configuration,
+) : Amplitude(configuration), AndroidNetworkListener.NetworkChangeCallback {
```

Comment:
why do we need this interface?

Reply:
`NetworkConnectivityChecker` is to check network proactively every time when track an event. `AndroidNetworkListener` is to listen network connection changes and flush when network comes back. If we don't have this interface, stored events will not be flushed automatically when network is back but can only be flushed when another event is tracked.

Reply:
I see. Remove `NetworkChangeCallback` interface from Amplitude in https://github.com/amplitude/Amplitude-Kotlin/pull/171/commits/70e5ff4ce0766363b78bc1733615cfde9f6a79b6